### DISCUSSION TITLE
Using Latex templates with Stencila

### DISCUSSION THREAD

**Discussion by finlay** [stencila/stencila#856] [General]
At Dragonfly we use a bunch of standard templates, including our own one. For example, this one:
https://files.dragonfly.co.nz/publications/pdf/Richardetal_2020_seabird-counts.pdf

The templates are available here:
https://github.com/dragonfly-science/dragonfly-latex-templates

We currently build them into .deb packages, and install them in Docker images for distribution. However, open to suggestions.

It would be great to be able to use custom latex templates.

**Comment by nokome** [comment]
Thanks for the suggestion @finlay. Please bare with me, I'm a LaTeX noobie so don't fully understand how these templates get used and by what tool...

Are the templates in `dragonfly-latex-templates` ones that you can specify in the `--template` option of Pandoc: https://pandoc.org/MANUAL.html#templates? That is, are these templates, in same sense as for example Jinja or Handlebars templates - a document that is passed to a template rendering engine along with a data context to produce a different  rendered result each time?

Or is it the case that, as seems common in LaTeX-land, "template" means something akin to "an example document with placeholder text that you can manually replace using an editor.

**Reply by finlay** [reply]
They are templates in the latex sense, not in the sense of pandoc.

More concretely, a document class is a special kind of latex package, with a `.cls` suffix. It essentially sets up the context for a document. 

some info here: http://www.texfaq.org/FAQ-clsvpkg

I think there is an option in pandoc to override the `\documentclass{}` declaration at the top of a latex document. Perhaps that is all that we need to do.

**Reply by edwardabraham** [reply]
I have also written templates for Pandoc so that output from markdown documents ends up in this format. All pretty straightforward.
